synapse/handlers/appservice.py

```diff
@@ -307,6 +314,23 @@ async def _notify_interested_services_ephemeral(
                             service, "presence", new_token
                         )
 
+                    elif (
+                        stream_key == "to_device_key"
+                        and self.msc2409_to_device_messages_enabled
+                    ):
+                        # Retrieve an iterable of to-device message events, as well as the
+                        # maximum stream token of the messages we were able to retrieve.
+                        events = await self._handle_to_device(service, new_token, users)
+                        if events:
```

Comment:
feels like this `if` condition should be inside `submit_ephemeral_events_for_as` rather than having to repeat it at every single call site.